Split development with Vietnam API

VietThan opened the pull request:
Related: #5

VietThan committed:
Split api development into VietnamAPI

VietnamAPI is worth split into a different project that others
might find more interested in than Quoc Hoi XV. VietnamAPI also
has more potential to be actually useful.

Using Github pages as a way to host the json

VietThan committed:
Merge remote-tracking branch 'origin/main' into separation

VietThan commented:
Used git submodule.

VietThan committed:
update VietnamAPI

VietThan committed:
Use fetch to retrieve json

On the suggestion of Khoa, using fetch to retrieve VietnamAPI
Succesfully populate provinces and districts dropdown.

Now need to work on districts to congressional units.

Issues: #5

VietThan commented:
@nguyenkhoa2407 review plz

nguyenkhoa2407 commented on .vscode/.ropeproject/config.py:
```diff
@@ -0,0 +1,114 @@
+
+
+def set_prefs(prefs):
+    """This function is called before opening the project"""
```
What does this file do :o

VietThan committed:
Populate results string of based on district data

Script will fetch data for province, districts and congressional
unit sequentially as users select dropdown options.

close: #8

VietThan commented:
This PR will:
- Split the development of data API with [VietnamAPI](github.com/VietThan/VietnamAPI) using git submodule
- Rewrote homepage js script to use fetch and rely on pulling data from VietnamAPI

The website is in version 1.0 with this PR.

nguyenkhoa2407 reviewed (approved):
looks good!
